packages/@aws-cdk/aws-rds/lib/engine.ts

```diff
@@ -0,0 +1,229 @@
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { compare } from './private/version';
+import { ParameterGroupFamily } from './props';
+
+abstract class BaseEngine {
+  /**
+   * The engine.
+   */
+  public readonly name: string;
+
+  /**
+   * The single user secret rotation application.
+   */
+  public readonly singleUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  /**
+   * The multi user secret rotation application.
+   */
+  public readonly multiUserRotationApplication: secretsmanager.SecretRotationApplication;
+
+  protected readonly parameterGroupFamilies?: ParameterGroupFamily[];
+
+  constructor(
+    name: string,
+    singleUserRotationApplication: secretsmanager.SecretRotationApplication,
+    multiUserRotationApplication: secretsmanager.SecretRotationApplication,
+    parameterGroupFamilies?: ParameterGroupFamily[]) {
```

Comment:
Since we're cleaning up this API for graduation, we should switch to the props model for constructors here - `DatabaseClusterEngineProps`, `BaseEngineProps`, etc.

Otherwise, adding new properties (as constructor args) without breaking existing API is going to be hard.

Reply:
Done.